BZ-46061: feat: Add Backend SDK for server-side payment integrations

prajwal-jp opened the pull request:
- Added Backend SDK support for server-side payment integration
    - Type-crafter added to generate backend types
    - Updated package.json to export ./backend
    - Refactored Rollup build for multiple entry points
    - Updated README with backend integration guide
    
<img width="1278" height="268" alt="Screenshot 2025-11-11 at 9 54 21 AM" src="https://github.com/user-attachments/assets/6dd15aef-d43a-40fa-bc77-54e83cc5afb2" />
<img width="1015" height="172" alt="image" src="https://github.com/user-attachments/assets/d0cd0b7c-110c-4d59-8f24-638abfbb5d08" />
<img width="1218" height="356" alt="image" src="https://github.com/user-attachments/assets/2ee97d9e-b923-4aa5-8a95-bbc21e9abb1b" />

sinha-sahil commented on .eslintignore:
```diff
@@ -3,6 +3,7 @@ dist/
 build/
 examples/
 coverage/
+src/generated/
```
lets include in eslint and lint and fix the generated output.

sinha-sahil commented on src/backend/index.ts:
```diff
@@ -0,0 +1,226 @@
+export type SDKCallback = (response: SDKResponse) => void;
+
+class BlazeSDK {
+  private static instance: BlazeSDK | null = null;
```
looks weird, we shouldn't store the instance inside the class itself.

sahil-sinha-jp commented on src/backend/index.ts:
```diff
@@ -0,0 +1,226 @@
+/**
```
Lets simplify this. callers should be outside of this exported class.

Avoid setup where users have to create an object of the class. lets stick to static class (like fe).
And, Lets name this something else like BlazeBackendSDK

prajwal-jp replied:
done

prajwal-jp committed:
BZ-46061: feat: Add Backend SDK for server-side payment integrations

- Added Backend SDK support for server-side payment integration
- Type-crafter added to generate backend types
- Updated package.json to export ./backend
- Refactored Rollup build for multiple entry points
- Updated README with backend integration guide
